challengeutils/evaluation_queue.py

```diff
@@ -1,45 +1,119 @@
+    (ie. 2020-02-21T23:53:27) to UTC epochtime in milliseconds and
+    timestring
 
     Args:
-        syn: Synpase object
-        name: Name of evaluation queue
-        description: Description of queue
-        parentid: Synapse project id
-        submissionInstructionsMessage: Instructions for submission
+        date_string: Local time date in YEAR-MM-DDTHH:MM:SS format
+                     (ie. 2020-02-21T23:53:27)
 
     Returns:
-        Evalation Queue
-    '''
-    evaluation = Evaluation(
-        name=name,
-        description=description,
-        contentSource=parentId,
-        submissionInstructionsMessage=submissionInstructionsMessage)
-    # submissionReceiptMessage="Thanks for submitting to %s!" % name)
-    queue = syn.store(evaluation)
-    return(queue)
-
-
-def set_evaluation_quota(syn, evalid, quota=3):
-    '''
-    Sets evaluation submission limit quota
-        {'firstRoundStart': u'2017-01-03T00:00:00.000Z',
-        'numberOfRounds': 1,
-        'roundDurationMillis': 3139200000,
-        'submissionLimit': 6}
+        dict: {"time_string": UTC time in YEAR-MM-DDTHH:MM:SS format
+               "epochtime_ms": UTC epoch time}
+
+    """
+    local_time_struct = time.strptime(date_string, "%Y-%m-%dT%H:%M:%S")
+    epochtime = time.mktime(local_time_struct)
+    utc_time_struct = time.gmtime(epochtime)
+    return {"time_string": time.strftime("%Y-%m-%dT%H:%M:%S.000Z",
+                                         utc_time_struct),
+            "epochtime_ms": int(epochtime*1000)}
+
+
+class SubmissionQuota:
+    """A SubmissionQuota object
+    https://rest-docs.synapse.org/rest/org/sagebionetworks/evaluation/model/SubmissionQuota.html
+    """
+    def __init__(self, firstRoundStart: str = None,
+                 roundDurationMillis: int = None, numberOfRounds: int = None,
+                 submissionLimit: int = None):
+        self.firstRoundStart = firstRoundStart
+        self.roundDurationMillis = roundDurationMillis
+        self.numberOfRounds = numberOfRounds
+        self.submissionLimit = submissionLimit
+
+
+def _create_quota(round_start: str = None, round_end: str = None,
+                  number_of_rounds: int = None, round_duration: int = None,
+                  submission_limit: int = None) -> SubmissionQuota:
+    """Creates the SubmissionQuota object
+
+    Args:
+        round_start: Start of round (local time) in YEAR-MM-DDTHH:MM:SS
+                     format (ie. 2020-02-21T17:00:00)
+        round_end: End of round (local time) in YEAR-MM-DDTHH:MM:SS format
+                   (ie. 2020-02-21T19:00:00)
+        number_of_rounds: Number of rounds
+        round_duration: Round duration in milliseconds
+        submission_limit: Number of submissions allowed per team
+
+    Returns:
+        SubmissionQuota object
+
+    """
+    if round_end is not None and round_duration is not None:
+        raise ValueError("Can only specify round_end or round_duration")
```

Comment:
Minor detail: exception uses `round_end`/`round_duration` whereas the CLI uses `round-end` and `round-duration`... maybe keep consistent?

Reply:
Nvm, saw the recent commit aha